playbooks/roles/common/tasks/repositories.yml

```diff
@@ -1,9 +1,21 @@
 ---
 - name: 'Setup Epel Repository'
   yum: name=http://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ release }}.noarch.rpm state=present
-- name: 'Setup Puppetlabs Repository'
-  yum: name=http://yum.puppetlabs.com/puppetlabs-release-el-{{ release }}.noarch.rpm state=present
+
+- name: 'Setup Puppet 3 Repository'
+  yum:
+    name: http://yum.puppetlabs.com/puppetlabs-release-el-{{ release }}.noarch.rpm
+    state: present
+  when: common_puppet_version is undefined or common_puppet_version == 3
+
+- name: 'Setup Puppet 4 Repository'
+  yum:
+    name: https://yum.puppetlabs.com/puppetlabs-release-pc1-el-{{ release }}.noarch.rpm
+    state: present
+  when: common_puppet_version is defined and common_puppet_version == 4
```

Comment:
I don't really understand ansible's way of working, it feels really kludgy to have so many small roles that may only do 1 small thing.  A common class seems obvious to me.

What do we do with EPEL? Another role?